app/code/Magento/Catalog/Model/ResourceModel/Product/Collection.php

```diff
@@ -2037,7 +2037,7 @@ protected function _applyZeroStoreProductLimitations()
             $this->getSelect()->join(
                 ['cat_pro' => $this->getTable('catalog_category_product')],
                 $joinCond,
-                ['cat_index_position' => 'position']
+                ['cat_index.position' => 'position']
```

Comment:
Fixed this now with last commit. You are correct, it was originally `cat_index_position` should be `cat_pro.position`.